### PR TITLE
chore(widgets): move widgets in centreon web package

### PR DESCRIPTION
**Pull request by kduret**
## Description

move widgets in centreon web package

image

**Fixes** MON-16861

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

### CODE REVIEW

File: centreon/www/install/createTables.sql [hunk 1]
```diff
@@ -2108,7 +2108,8 @@ CREATE TABLE `widget_models` (
   `title` varchar(255) NOT NULL,
   `description` TEXT NOT NULL,
   `url` varchar(255) NOT NULL,
-  `version` varchar(255) NOT NULL,
+  `is_internal` enum('0','1') NOT NULL DEFAULT '0',
```
Comment (kduret) [inline centreon/www/install/createTables.sql:2111]:
```suggestion
  `is_internal` BOOLEAN NOT NULL DEFAULT '0',
```

File: .github/workflows/get-version.yml [hunk 2]
```diff
@@ -116,7 +116,7 @@ jobs:
           echo "stability=$STABILITY" >> $GITHUB_OUTPUT
           echo "env=$MAJOR-$ENV" >> $GITHUB_OUTPUT
 
-          DOCKERIZE_MATRIX='["alma9"]'
+          DOCKERIZE_MATRIX='["alma8","alma9","bullseye"]'
```
Comment (kduret) [inline .github/workflows/get-version.yml:119]:
```suggestion
          DOCKERIZE_MATRIX='["alma9"]'
```

File: centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php [hunk 3]
```diff
@@ -69,14 +69,16 @@ public function getModulesVersion(): array
...
         $versions = [];
 
-        $widgetModelsRequest = $this->translateDbName('SELECT `title`, `version` FROM `:db`.widget_models');
+        $widgetModelsRequest = $this->translateDbName('SELECT `title`, `version`, `is_internal` FROM `:db`.widget_models');
         if (($statement = $this->db->query($widgetModelsRequest)) !== false) {
             while ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
-                $versions[(string) $result['title']] = (string) $result['version'];
+                $versions[(string) $result['title']] = $result['is_internal'] === 1
+                    ? (string) $result['version']
+                    : $webVersion;
```
Comment (kduret) [inline centreon/src/Centreon/Infrastructure/Platform/PlatformRepositoryRDB.php:81]:
```suggestion
                    ? $webVersion
                    :  (string) $result['version'];
```

**Review by a-launois — DISMISSED** [review]

**Review by arnapou — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by arnapou — APPROVED** [review]

**Review by sc979 — APPROVED** [review]
LGTM on pipeline part

**Review by tuntoja — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**